src/test/java/test/project/bookshop/service/BookServiceImplTests.java

```diff
@@ -0,0 +1,253 @@
+package test.project.bookshop.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.jpa.domain.Specification;
+import test.project.bookshop.dto.book.BookDto;
+import test.project.bookshop.dto.book.BookRequestDto;
+import test.project.bookshop.dto.book.BookSearchParametersDto;
+import test.project.bookshop.dto.book.BookWithoutCategoryIdsDto;
+import test.project.bookshop.exception.EntityNotFoundException;
+import test.project.bookshop.mapper.BookMapper;
+import test.project.bookshop.model.Book;
+import test.project.bookshop.model.Category;
+import test.project.bookshop.repository.book.BookRepository;
+import test.project.bookshop.repository.book.BookSpecificationBuilder;
+import test.project.bookshop.service.impl.BookServiceImpl;
+
+@ExtendWith(MockitoExtension.class)
+class BookServiceImplTests {
+    private static final Long FIRST_BOOK_ID = 1L;
+    private static final Long SECOND_BOOK_ID = 2L;
+    private static final Long NON_EXISTENT_BOOK_ID = 999L;
+    private static final PageRequest PAGEABLE = PageRequest.of(0, 10);
+
+    @Mock
+    private BookSpecificationBuilder bookSpecificationBuilder;
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private BookMapper bookMapper;
+    @Mock
+    private CategoryService categoryService;
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    @Test
+    @DisplayName("Save a new book successfully")
+    void save_NewBook_ReturnsSavedBookDto() {
```

Comment:
I recommend you use Given-When-Then construction and increase the readability of your solution.